internal/debug/transport.go

```diff
@@ -1,38 +1,72 @@
 package debug
 
 import (
+	"bytes"
+	"fmt"
 	"io"
 	"net/http"
+	"net/http/httptrace"
 	"net/http/httputil"
 )
 
 // Transport implements http.RoundTripper and can be used to wrap other HTTP
-// transports to dump request and responses for debugging.
+// transports for debugging, normally http.DefaultTransport.
 type Transport struct {
 	http.RoundTripper
 	Output io.Writer
+	// Dump controls whether to dump HTTP request and responses.
+	Dump bool
+	// Trace enables usage of net/http/httptrace.
+	Trace bool
 }
 
 func (t *Transport) RoundTrip(req *http.Request) (*http.Response, error) {
-	b, err := httputil.DumpRequestOut(req, true)
-	if err != nil {
-		return nil, err
+	var buf bytes.Buffer
```

Comment:
Maybe rename `buf` to something more explicit? I had hard time follow between `b` and `buf` interchangeably.
`debugLogBuffer`? `logBuffer`?